### ISSUE TITLE
Incorrect registration flash message when email verification is enabled

### ISSUE THREAD

**Issue by piotrpog** [craftcms/cms#5213]
If email verification is enabled, after user registers using frontend form, "user registered" flash message is still displayed - instead of something like "please check your email to confirm account".

**Comment by brandonkelly** [comment]
Technically they were registered successfully so the notice is accurate, but perhaps we should just start letting the form define what the success message should be, similar to what we recently did for Craft Commerce: https://github.com/craftcms/commerce/commit/427ddb009dda6dc8f151755dd0c529291a29a42d

**Comment by piotrpog** [comment]
How can i display some additional message then? I tried this code in craftmodule but it didnt worked

```
Event::on(
    User::class,
    User::EVENT_AFTER_SAVE,
    function(ModelEvent $event) {
          Craft::$app->getSession()->setNotice('x');
    }
);
```

**Comment by brandonkelly** [comment]
In the meantime the easiest way to do it would be using [static message translations](https://docs.craftcms.com/v3/static-translations.html). For example create a `translations/<language>/app.php` file (replace `<language>` with your site’s language code, e.g. `en-US`), and give it this code:

```php
<?php

return [
    'User registered.' => 'Please check your email to confirm account.',
];
```

**Comment by brandonkelly** [comment]
This is now resolved for the next 3.4 release. Going forward you will be able to customize the success notice on your user registration forms by including this param:

```html
<input type="hidden" name="userRegisteredNotice" value="Please check your email to confirm account ">
```

**Comment by brandonkelly** [comment]
That `userRegisteredNotice` has been deprecated for the next release (3.4.24), in favor of a new hashed `successMessage` param:

```twig
{{ hiddenInput('successMessage', 'Custom success message'|hash) }}
```

`userRegisteredNotice` will be fully removed in Craft 3.5 due to a security concern.
